relentless/simulate/lammps.py

```diff
@@ -0,0 +1,361 @@
+
+        # make box from ensemble
+        box = self.extract_box_params(sim)
+        if not np.all(np.isclose(box[-3:],0)):
+            cmds += ['region box prism {} {} {} {} {} {} {} {} {}'.format(*box)]
+        else:
+            cmds += ['region box block {} {} {} {} {} {}'.format(*box[:-3])]
+        cmds += ['create_box {N} box'.format(N=len(sim.ensemble.types))]
+
+        # use lammps random initialization routines
+        for i in sim.ensemble.types:
+            cmds += ['create_atoms {typeid} random {N} {seed} box'.format(typeid=sim.type_map[i],
+                                                                          N=sim.ensemble.N[i],
+                                                                          seed=self.seed+sim.type_map[i]-1)]
+        cmds += ['mass * 1.0',
+                 'velocity all create {temp} {seed}'.format(temp=sim.ensemble.T,
+                                                            seed=self.seed)]
+
+        cmds += self.attach_potentials(sim)
+
+        return cmds
+
+class MinimizeEnergy(LAMMPSOperation):
+    def __init__(self, energy_tolerance, force_tolerance, max_iterations, dt):
+        self.energy_tolerance = energy_tolerance
+        self.force_tolerance = force_tolerance
+        self.max_iterations = max_iterations
+
+    def to_commands(self, sim):
+        cmds = ['minimize {etol} {ftol} {maxiter} {maxeval}'.format(etol=self.energy_tolerance,
+                                                                    ftol=self.force_tolerance,
+                                                                    maxiter=self.max_iterations,
+                                                                    maxeval=100*self.max_iterations)]
+
+        return cmds
+
+class AddLangevinIntegrator(LAMMPSOperation):
+    def __init__(self, dt, friction, seed):
+        self.friction = friction
+        self.seed = seed
+
+        self._fix_langevin = self.add_fix()
+        self._fix_nve = self.add_fix()
+
+    def to_commands(self, sim):
+        # obtain per-type mass
+        mass_per_type = {}
+
+        mass_ptr = sim.lammps.extract_atom('mass',2)
+        Ntypes = len(sim.ensemble.types)
+        mass = np.ctypeslib.as_array(mass_ptr,(Ntypes+1,))[1:]
+
+        types_ptr = sim.lammps.extract_atom('type',0)
+        Natoms = sum(sim.ensemble.N.todict().values())
+        types = np.unique(np.ctypeslib.as_array(types_ptr,(Natoms,)))
+        types = [str(t) for t in types]
+
+        for t,m in zip(types,mass):
+            mass_per_type[t] = m
+
+        # obtain per-type friction factor
+        if np.isscalar(self.friction):
+            friction_per_type = {}
+            for t in types:
+                friction_per_type[t] = self.friction
+        else:
+            if sorted(self.friction.keys()) != sorted(types):
+                return ValueError('The friction factor types must match the atom types.')
+            friction_per_type = self.friction
```

Comment:
You might want to do this in a lazy way like I did in HOOMD so that you don't need as much code:
```py
friction = np.zeros_like(mass)
for t in sim.ensemble.types:
    try:
        friction[sim.type_map[t]] = self.friction[t]
    except KeyError:
        friction[sim.type_map[t]] = self.friction
```